ksp_plugin/pile_up.cpp

```diff
@@ -419,44 +426,153 @@ void PileUp::DeformPileUpIfNeeded(Instant const& t) {
+      Identity<ApparentPileUp, NonRotatingPileUp>()(apparent_angular_momentum),
+      angular_momentum_));
+
+  // We apply a rigid rotational correction to the motions of the parts coming
+  // from the game (the apparent motions) so as to enforce the conservation of
+  // the angular momentum (|angular_momentum_| is authoritative).
+  // Mapping L_apparent to L_actual by a rigid motion can be done in many ways.
+  // We prefer doing some of that correction by a change of attitude, rather
+  // than a solely by a change in angular velocity, since, under isotropic
+  // conditions, a change in attitude does not alter the physical system (and in
+  // particular it does not mess with symplectic integration).  We thus try to
+  // map L̂_apparent to L̂_actual, by rotation around the correction axis defined
+  // above, which is perpendicular to both, and to impart the appropriate
+  // angular velocity correction to correct the norm of L.
+  // This amounts to trusting the direction of the angular momentum with respect
+  // to the vessel as given to us by the game.
+  // However, mapping L̂_apparent to L̂_actual is essentially singular when either
+  // is 0.  We remedy to that by only performing an attitude correction if its
+  // angle would be less than ω Δt, where ω is the smaller of the two equivalent
+  // angular frequencies |L_actual / I| and |L_apparent / I|.
+  // As a result, as either L tends towards 0, so does the greatest attitude
+  // correction that we allow.
+  // If the attitude correction would exceed this threshold, we leave the
+  // attitude unchanged, and the correction in angular momentum is effected
+  // solely by a change in angular velocity.
+
+  // The correction is computed via an intermediate frame.
+  // In the |EquivalentRigidPileUp| reference frame, a rigid body with the same
+  // inertia and angular momentum as the pile up would be immobile.
+  // If no attitude correction is performed, the axes of
+  // |EquivalentRigidPileUp|, |ApparentPileUp|, and |NonRotatingPileUp| are
+  // identical.
+  // If an attitude correction is performed, the y axis of
+  // |EquivalentRigidPileUp| is the direction of the angular momentum, and the x
+  // axis is the correction axis.
+  // NOTE(egg): while the correction axis is essentially singular at
+  // L_apparent = L_actual, the correction itself is only removably singular (as
+  // the angle goes to 0).  Since the computation ensured that
+  // |apparent_correction_axis.coordinates() ==
+  //  actual_correction_axis.coordinates()| exactly, we need not worry about the
+  // intermediate essential singularity, and only need to remove it.
+  using EquivalentRigidPileUp = Frame<enum class EquivalentRigidPileUpTag>;
+
+  auto const L̂_apparent = NormalizeOrZero(apparent_angular_momentum);
+  auto const L̂_actual = NormalizeOrZero(angular_momentum_);
+  // NOTE(egg):
+  // — The two sides of this disjunction are always equal;
```

Comment:
Should this be checked?

Reply:
If we want to check something it is the equality of the `coordinates` commented above, not a random consequence thereof.